src/notation/view/notationpaintview.cpp

```diff
@@ -776,6 +807,16 @@ RectF NotationPaintView::toLogical(const QRect& rect) const
     return RectF::fromQRectF(m_matrix.inverted().mapRect(scaledRect));
 }
 
+RectF NotationPaintView::fromLogical(const QRect& rect) const
+{
+    double scale = guiScaling();
+
+    RectF mappedRect = RectF::fromQRectF(m_matrix.mapRect(rect));
+
+    mappedRect.setBottomRight(mappedRect.bottomRight() / scale);
```

Comment:
Same comments as above, move `double scale = guiScaling()` a few lines down or remove the variable `scale` completely.